torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -0,0 +1,392 @@
+ROOT_MODULE_DICT_KEY = "root_module"
+QAT_MODULE_DICT_KEY = "qat_module"
+REFERENCE_QUANTIZED_MODULE_DICT_KEY = "reference_quantized_module_for_root"
+FUSER_MODULE_DICT_KEY = "fuser_module"
+FUSER_METHOD_DICT_KEY = "fuser_method"
+ROOT_NODE_GETTER_DICT_KEY = "root_node_getter"
+EXTRA_INPUTS_GETTER_DICT_KEY = "extra_inputs_getter"
+NUM_TENSOR_ARGS_TO_OBSERVATION_TYPE_DICT_KEY = "num_tensor_args_to_observation_type"
+INPUT_TYPE_TO_INDEX_DICT_KEY = "input_type_to_index"
+INPUT_OUTPUT_OBSERVED_DICT_KEY = "input_output_observed"
+OVERWRITE_OUTPUT_FAKE_QUANTIZE_DICT_KEY = "overwrite_output_fake_quantize"
+OVERWRITE_OUTPUT_OBSERVER_DICT_KEY = "overwrite_output_observer"
+
+
+@dataclass
+class DTypeConfig:
+    """
+    Data type config for ops defined in :class:`~torch.ao.quantization.backend_config.BackendConfig`.
+    """
+    input_dtype: Optional[torch.dtype] = None
+    output_dtype: Optional[torch.dtype] = None
+    weight_dtype: Optional[torch.dtype] = None
+    bias_dtype: Optional[torch.dtype] = None
+    is_dynamic: Optional[bool] = None
+
+    @classmethod
+    def from_dict(cls, dtype_config_dict: Dict[str, Any]) -> DTypeConfig:
+        """
+        Create a `DTypeConfig` from a dictionary with the following items (all optional):
+
+            "input_dtype": torch.dtype
+            "output_dtype": torch.dtype
+            "weight_dtype": torch.dtype
+            "bias_type": torch.dtype
+            "is_dynamic": bool
+        """
+        input_dtype = dtype_config_dict.get(INPUT_DTYPE_DICT_KEY, None)
+        output_dtype = dtype_config_dict.get(OUTPUT_DTYPE_DICT_KEY, None)
+        weight_dtype = dtype_config_dict.get(WEIGHT_DTYPE_DICT_KEY, None)
+        bias_dtype = dtype_config_dict.get(BIAS_DTYPE_DICT_KEY, None)
+        is_dynamic = dtype_config_dict.get(IS_DYNAMIC_DICT_KEY, None)
+        return cls(input_dtype, output_dtype, weight_dtype, bias_dtype, is_dynamic)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert this `DTypeConfig` to a dictionary with the items described in
+        :func:`~torch.ao.quantization.backend_config.DTypeConfig.from_dict`.
+        """
+        dtype_config_dict: Dict[str, Any] = {}
+        if self.input_dtype is not None:
+            dtype_config_dict[INPUT_DTYPE_DICT_KEY] = self.input_dtype
+        if self.output_dtype is not None:
+            dtype_config_dict[OUTPUT_DTYPE_DICT_KEY] = self.output_dtype
+        if self.weight_dtype is not None:
+            dtype_config_dict[WEIGHT_DTYPE_DICT_KEY] = self.weight_dtype
+        if self.bias_dtype is not None:
+            dtype_config_dict[BIAS_DTYPE_DICT_KEY] = self.bias_dtype
+        if self.is_dynamic is not None:
+            dtype_config_dict[IS_DYNAMIC_DICT_KEY] = self.is_dynamic
+        return dtype_config_dict
+
+
+class BackendConfig:
+    # TODO: refer to native fbgemm BackendConfig once that is implemented
+    """
+    Config that defines which ops are supported and how they are quantized on a custom backend.
```

Comment:
guess we need to rephrase this description now, it does not talk about ops, it's about "q/dq placement rules that are supported on this backend, the rules are described in terms of how we place q/dq ops for input and output of some patterns"

Reply:
Ok. Should we say `quant/dequant ops` instead of `q/dq`?

Reply:
yeah makes sense